### ISSUE TITLE
Cucumber: Report broken when using Example setps

### ISSUE THREAD

**Issue by asvdw** [serenity-bdd/serenity-core#1685]
Hello,
Using serenity version 2.0.40 and cucumber 1.9.37, there are unprocessed placeholders in examples table when executing set of example.

ie with this feature file:
```

Feature: Lookup a definition with sets
  In order to talk better
  As an English student
  I want to look up word definitions

  Scenario Outline: Looking up the definition of 'apple' with sets
    Given the user is on the Wikionary home page
    When the user looks up the definition of the word '<word>'
    Then they should see the definition '<definition>'

  @Daily
    Examples: representative set
      | word  | definition                                                                                    |
      | apple | A common, round fruit produced by the tree Malus domestica, cultivated in temperate climates. |


    Examples: other set
      | word | definition                                                                                     |
      | pear | An edible fruit produced by the pear tree, similar to an apple but elongated towards the stem. |

```
and `mvn verify -Dtags=Daily', tests are rightly executed (look for apple but not for pear) but in the report you'll see something like this, in Specifications tab

word | definition |  
-- | -- | --
apple | A common, round fruit produced by the tree Malus domestica, cultivated in temperate climates. | {example-result:Lookup a definition with sets!Looking up the definition of 'apple' with sets[0][0]}

Furthermore, if I run `mvn serenity:aggregate` _after_ `mvn clean verify`, placeholders for all examples are now replaced by "success" icon, even if they has not bu run - in this case, pear and apple will be marked as successful.

Any idea ?
Thank you

**Comment by wakaleo** [comment]
This will be tricky to fix.

**Comment by asvdw** [comment]
Hello,
In fact... it's tricky ;) 

Firstly I have to clarify the issue:
`mvn serenity:aggregate` will replace placeholders, but with inaccurate results
`mvn serenity:aggregate -Dtags=Daily` will leave placeholders 

The problem is about tags.

then I saw that:
in `HtmlAggregateStoryReporter.generateReportsForTestResultsFrom(File sourceDirectory)`, 
`allTestOutcomes.withTags(this.getTags())` remove all dataTable rows, except the first one

So commenting this as a quick and dirty fix does the trick for my issue, but there may be some side effects:
```
  //      if (!StringUtils.isEmpty(this.tags)) {
  //          allTestOutcomes = allTestOutcomes.withTags(this.getTags());
  //      }
```

I do not have the time to look further right now, but that seems to be the right track...

**Comment by jcunliffe1** [comment]
Using multiple tags with subcategories, which then are split via tagtype definition, is also broken on example data tags.

**Comment by wakaleo** [comment]
@jcunliffe1 can you provide an example project?

**Comment by wakaleo** [comment]
Fixed in 3.2.0
